### PR TITLE
When a session is created, raise exception if HTTP error response (#41)

### PR DESCRIPTION
**Pull request by hklang10**
This patch raises an immediate exception with the full IG error message.  Currently the exception raised is "key error : cst" during _set_headers().

### CODE REVIEW

File: trading_ig/rest.py [hunk 1]
```diff
@@ -76,6 +76,11 @@ def _create_first(self, endpoint, params, session):
         response = session.post(url,
                                 data=json.dumps(params),
                                 headers=self.HEADERS['BASIC'])
+        if response.ok:
```
Comment (femtotrader) [inline trading_ig/rest.py]: Why not using:

```python
    if not response.ok:
        ...
```

Comment (hklang10) [reply trading_ig/rest.py]: Just the way I was thinking ! - I will make an update.

**Comment by hklang10** [comment]
@femtotrader is it correct here to `raise IGException` and not use standard Exception?

**Comment by femtotrader** [comment]
raise `IGException` and not use standard Exception is correct.

In fact it just depend how (if you) you will catch this exception.

**Comment by femtotrader** [comment]
Thanks @hklang10

**Pull request merged**